rest-server/src/templates/dockerContainerScript.mustache

```diff
@@ -67,20 +67,24 @@ if [ "$(cat ContainerList | wc -l)" -ne $PAI_TASKS_NUM ]; then
   printf "%s %s\n%s\n%s\n\n" \
     "[ERROR]" "ContainerList" \
     "$(cat ContainerList)" \
-    "$(cat ContainerList | wc -l) containers are available, less than $PAI_TASKS_NUM, exit ..."
+    "$(cat ContainerList | wc -l) containers are available, not equal to $PAI_TASKS_NUM, exit ..."
   exit 2
 fi
 
 export PAI_TASK_ROLE_INDEX=$((`cat ContainerList | grep "^$PAI_TASK_ROLE_NO-" | grep -n "$PAI_TASK_ROLE_NO-$PAI_TASK_INDEX-$PAI_CURRENT_CONTAINER_IP-$PAI_CURRENT_CONTAINER_PORT" | cut -d ":" -f 1`-1))
 for i in `seq 0 $((PAI_TASK_ROLES_NUM-1))`; do
   export PAI_TASK_ROLE_${i}_HOST_LIST=`cat ContainerList | grep "^$i-" | cut -d "-" -f 3-4 | tr "-" ":" | sed -e :E -e "N;s/\n/,/;tE"`
 done
+rm ContainerList
 
 printf "%s %s\n%s\n\n" "[INFO]" "ENV" "$(printenv | sort)"
 
-cd $HOME
-hdfs dfs -get $PAI_CODE_DIR code || exit 1
-cd code
+if [[ -z $PAI_CODE_DIR ]]; then
+  cd $HOME
+  hdfs dfs -get $PAI_CODE_DIR code || exit 1
+  cd code
+fi
+
```

Comment:
One thing we need to discuss here:

If user specifies code, then we goto "$WORKDIR/code", else we just stay at "$WORKDIR". This is inconsistent. Also the subfolder "code" is created by us implicitly, this might conflict with user container folder structure.

A possible solution is let user determine their code folder name entirely. For example, suppose user specified "hdfs://xxx/nni", we will download nni to "$WORKDIR/nni". Also we don't cd into that folder, just keep $WORKDIR unchanged, so it's user's duty to input command line like "nni/start.sh". By this way we can keep consistency if user specified codeDir or not.

Reply:
updated